packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import { CrossL2Inbox } from "src/L2/CrossL2Inbox.sol";
+import { Encoding } from "src/libraries/Encoding.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/IL2ToL2CrossDomainMessenger.sol";
+import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
```

Comment:
Can we reduce deps on oz here and use a simple 1153 based reentrency guard?

Reply:
This keeps the storage layout clean, saves a small amount of gas and transient storage is better for running in parallel evm

Reply:
can we just use the new `TransientContext` library?

Reply:
Can we implement a simple tstore based reentrency guard?

Reply:
what if we just used `ENTERED_SLOT` for reentrancy guard of the function as a whole?